quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -0,0 +1,1120 @@
+            .map_err(|err| MetastoreError::InternalError {
+                message: "Failed to get connection".to_string(),
+                cause: anyhow::anyhow!(err),
+            })?;
+
+        // Check for the existence of index.
+        let index_exists: bool =
+            self.is_index_exist(&conn, index_id)
+                .map_err(|err| MetastoreError::InternalError {
+                    message: "Failed to publish splits".to_string(),
+                    cause: anyhow::anyhow!(err),
+                })?;
+        if !index_exists {
+            return Err(MetastoreError::IndexDoesNotExist {
+                index_id: index_id.to_string(),
+            });
+        }
+
+        let list_splits_statement = if let Some(time_range) = time_range_opt {
+            schema::splits::dsl::splits
+                .filter(
+                    schema::splits::dsl::index_id
+                        .eq(index_id)
+                        .and(schema::splits::dsl::split_state.eq(state.to_string()))
+                        .and(
+                            schema::splits::dsl::end_time_range.is_null().or(
+                                schema::splits::dsl::end_time_range
+                                    .ge(time_range.start)
+                                    .and(schema::splits::dsl::start_time_range.lt(time_range.end)),
+                            ),
+                        ),
+                )
+                .into_boxed()
+        } else {
+            schema::splits::dsl::splits
+                .filter(
+                    schema::splits::dsl::index_id
+                        .eq(index_id)
+                        .and(schema::splits::dsl::split_state.eq(state.to_string())),
+                )
+                .into_boxed()
+        };
+        debug!(sql=%debug_query::<Pg, _>(&list_splits_statement).to_string());
+        let model_splits: Vec<model::Split> =
+            list_splits_statement
+                .load(&conn)
+                .map_err(|err| MetastoreError::InternalError {
+                    message: "Failed to select splits".to_string(),
+                    cause: anyhow::anyhow!(err),
+                })?;
+
+        // Make the split metadata and footer offsets from database model.
+        let mut split_metadata_footer_offset_list: Vec<SplitMetadataAndFooterOffsets> = Vec::new();
+        for model_split in model_splits {
+            let split_metadata_and_footer_offsets =
+                match model_split.make_split_metadata_and_footer_offsets() {
+                    Ok(metadata) => metadata,
+                    Err(err) => {
+                        error!("Failed to make split metadata and footer offsets {:?}", err);
+                        continue;
+                    }
+                };
+
+            // The tags are empty (unspecified) or the tags are specified
+            // and the split has a matching tag, it will be added to the list.
+            let mut match_tag = false;
```

Comment:
Can you that logic in an external function?

/// ...
fn does_match_tag_filter(tags: &[String], tag_filter: &[String]) -> bool {
}


it could be factorized in the single file metastore too.

Reply:
https://github.com/quickwit-inc/quickwit/blob/8b460dfd684912e3668f8192e5a9d96edd4fbcaa/quickwit-metastore/src/metastore/postgresql_metastore.rs#L811-L819

https://github.com/quickwit-inc/quickwit/blob/8b460dfd684912e3668f8192e5a9d96edd4fbcaa/quickwit-metastore/src/metastore/single_file_metastore.rs#L386-L393